### DISCUSSION TITLE
Track clicks intercept link on plain_body, but not in html_body

### DISCUSSION THREAD

**Discussion by dragoangel** [postalserver/postal#1437] [Help with using Postal]
Here is example of powershell test script which send email:
```
$Server = 'https://postal.example.com'
$APIKey = 'XXXX'

$Function = '/api/v1/send/message'
$Method = 'Post'
$ContentType = 'application/json'

$From = 'sender@example.com'
$To = 'recipient@example.org'
$Subject = 'Test'
$PlainBody = @'
test
-- 
*Best Regards,*
Name
DevOps Engineer
Example <https://example.com/>
E-mail: sender@example.com <mailto:sender@example.com>
'@
$HtmlBody = @'
<html>
  <head>
    <meta http-equiv="content-type" content="text/html; charset=UTF-8">
  </head>
  <body>
    test<br>
    <div class="signature">-- <br>
      <b>Best Regards,</b><br>
      Name<br>
      DevOps Engineer<br>
      <a href="https://example.com/">Example</a><br>
      E-mail: <a href="mailto:sender@example.com">sender@example.com</a></div>
  </body>
</html>
'@

$ToArray = @(${To})

$Headers = @{
    'X-Server-API-Key' = ${APIKey};
}

$Body = @{
    'from' = ${From};
    'to' = ${ToArray};
    'subject' = ${Subject};
    'plain_body' = ${PlainBody};
    'html_body' = ${HtmlBody};
}

Invoke-RestMethod -Uri ${Server}${Function} -Method ${Method} -Headers ${Headers} -Body (${Body}|ConvertTo-Json) -ContentType $ContentType
```
Result:
```
----==_mimepart_5ef47035eea67_2df03318330c840b
Content-Type: text/plain;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

test
-- 
*Best Regards,*
Name
DevOps Engineer
Example <http://click.example.com/some/id/>
E-mail: sender@example.com <mailto:sender@example.com>
----==_mimepart_5ef47035eea67_2df03318330c840b
Content-Type: text/html;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

<html>
  <head>
    <meta http-equiv="content-type" content="text/html; charset=UTF-8">
  </head>
  <body>
    test<br>
    <div class="signature">-- <br>
      <b>Best Regards,</b><br>
      Name<br>
      DevOps Engineer<br>
      <a href="https://example.com/">Example</a><br>
      E-mail: <a href="mailto:sender@example.com">sender@example.com</a></div>
  </body>
</html>
----==_mimepart_5ef47035eea67_2df03318330c840b--
```
As you can see  `<a href="https://example.com/">Example</a><br>` wasn't been changed to ` <a href="http://click.example.com/some/id">Example</a><br>`

**Comment by dragoangel** [comment]
Also not send `plain_body` at all - not fix this too.

**Comment by willpower232** [comment]
My first guess would be related to how the string or file is encoded (i.e. UTF8 or otherwise), I haven't had a chance to diagnose a similar problem we're seeing yet.

**Comment by dragoangel** [comment]
Hi @willpower232 I provided sample script which use windows PowerShell to send sample email, need only change host and key and from & to. There no files or attachments, all data provided by POST body & headers. This:
`
'plain_body' = ${PlainBody};
'html_body' = ${HtmlBody};
`
parameters encoded in one request body, they both has same encoding. URLs in plaintext was changed to trackdomain correctly, but html version links wasn't been changed. Do I need to use some special syntax in HTML body? As I understand from reading docs - no, that's why I created this ticket. Thank you for helping in advance.

**Comment by arnebr** [comment]
I have the same/similiar issue, using the PHP library. Links and tracking is not added but the tracking domain is working.

_Edit_: When I created a new mailserver it did work with the link tracking. Did I miss a setting?
_Edit 2_: Recreated the same server with the same domain, still doesn't work. I noticed one difference though, I created one tracking domain from the organization domain and the other from the mail server domain.

**Comment by dragoangel** [comment]
@arnebr that what you mentioned about company level & server level is default way to handle it. It done to easier manage many domains setup over simple cnames, this fine:

- you have one main domain
- all other domains are simple cname to main one

Issue that future is broken 😩 particularly.
